plugin/core/settings.py

```diff
@@ -15,11 +16,17 @@ class ClientConfigs:
     def __init__(self) -> None:
         self.all: dict[str, ClientConfig] = {}
         self.external: dict[str, ClientConfig] = {}
-        self._listener: Callable[[str | None], None] | None = None
+        self._clients_listener: Callable[[str | None], None] | None = None
+        self._userprefs_listener: Callable[[], None] | None = None
+        self._clients_hash: int | None = None
 
     def _notify_listener(self, config_name: str | None = None) -> None:
```

Comment:
Would be good to rename to `_notify_clients_listener` for consistency.

Reply:
https://github.com/sublimelsp/LSP/pull/2448/commits/3dd85ec876ab1ccf2288cff8a22f5b951a88fc0e